vochain/app.go

```diff
@@ -74,6 +81,9 @@ type BaseApplication struct {
 	dataDir             string
 	genesisInfo         *tmtypes.GenesisDoc
 
+	// prepareProposalLock is used to avoid concurrent calls between PrepareProposal and FinalizeBlock
```

Comment:
Given that is also used in ProcessProposal, is it right to understand that we try to avoid concurrency issues between PrepareProposal, ProcessProposal and FinalizeBlock ?

Reply:
yes